Compilation warnings on Windows

There are some warnings at Azure Pipelines on Windows machine for `master` branch:

```
c:\program files (x86)\microsoft visual studio\2017\enterprise\vc\tools\msvc\14.15.26726\include\xutility(2903): warning C4244: '=': conversion from 'const _Ty' to 'float', possible loss of data [D:\a\1\s\build\_lightgbm.vcxproj]
          with
          [
              _Ty=int
          ] (compiling source file D:\a\1\s\src\boosting\boosting.cpp)
  c:\program files (x86)\microsoft visual studio\2017\enterprise\vc\tools\msvc\14.15.26726\include\xutility(2917): note: see reference to function template instantiation 'void std::_Fill_unchecked1<_FwdIt,_Ty>(_FwdIt,_FwdIt,const _Ty &,std::false_type)' being compiled
          with
          [
              _FwdIt=float *,
              _Ty=int
          ] (compiling source file D:\a\1\s\src\boosting\boosting.cpp)
  c:\program files (x86)\microsoft visual studio\2017\enterprise\vc\tools\msvc\14.15.26726\include\xutility(2925): note: see reference to function template instantiation 'void std::_Fill_unchecked<float*,_Ty>(_FwdIt,_FwdIt,const _Ty &)' being compiled
          with
          [
              _Ty=int,
              _FwdIt=float *
          ] (compiling source file D:\a\1\s\src\boosting\boosting.cpp)
  d:\a\1\s\src\boosting\rf.hpp(82): note: see reference to function template instantiation 'void std::fill<std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>,int>(_FwdIt,_FwdIt,const int &)' being compiled
          with
          [
              _Ty=LightGBM::label_t,
              _FwdIt=std::_Vector_iterator<std::_Vector_val<std::_Simple_types<LightGBM::label_t>>>
          ] (compiling source file D:\a\1\s\src\boosting\boosting.cpp)
```

The full log can be found here: https://lightgbm-ci.visualstudio.com/lightgbm-ci/_build/results?buildId=172&view=logs in `Windows regular` job

ping @guolinke, @chivee

--- a/src/boosting/rf.hpp
+++ b/src/boosting/rf.hpp
@@ -47,7 +47,7 @@ public:
       tmp_grad_.resize(num_data_);
       tmp_hess_.resize(num_data_);
     }
-    tmp_score_.resize(num_data_, 0.0f);
+    tmp_score_.resize(num_data_, 0.0);
   }
 
   void ResetConfig(const Config* config) override {
@@ -74,31 +74,31 @@ public:
       tmp_grad_.resize(num_data_);
       tmp_hess_.resize(num_data_);
     }
-    tmp_score_.resize(num_data_, 0.0f);
+    tmp_score_.resize(num_data_, 0.0);
   }
 
   void GetRFTargets(const Dataset* train_data) {
     auto label_ptr = train_data->metadata().label();
-    std::fill(hessians_.begin(), hessians_.end(), 1);
+    std::fill(hessians_.begin(), hessians_.end(), 1.0f);
     if (num_tree_per_iteration_ == 1) {
       OMP_INIT_EX();
       #pragma omp parallel for schedule(static,1)
       for (data_size_t i = 0; i < train_data->num_data(); ++i) {
         OMP_LOOP_EX_BEGIN();
-        double label = label_ptr[i];
+        score_t label = label_ptr[i];
         gradients_[i] = static_cast<score_t>(-label);
         OMP_LOOP_EX_END();
       }
       OMP_THROW_EX();
     }
     else {
-      std::fill(gradients_.begin(), gradients_.end(), 0);
+      std::fill(gradients_.begin(), gradients_.end(), 0.0f);
       OMP_INIT_EX();
       #pragma omp parallel for schedule(static,1)
       for (data_size_t i = 0; i < train_data->num_data(); ++i) {
         OMP_LOOP_EX_BEGIN();
-        double label = label_ptr[i];
-        gradients_[i + static_cast<int>(label) * num_data_] = -1;
+        score_t label = label_ptr[i];
+        gradients_[i + static_cast<int>(label) * num_data_] = -1.0f;
         OMP_LOOP_EX_END();
       }
       OMP_THROW_EX();

